### PR TITLE
[MetalFX] Add the framework (which was added in Xcode 14) and add xcode 15 support.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by haritha-mohan — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

File: src/metalfx.cs [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+		float InputContentMaxScale { get; set; }
+
+		[Export ("newTemporalScalerWithDevice:")]
+		[return: NullAllowed, Release]
+		IMTLFXTemporalScaler Create (IMTLDevice device);
+
+		[Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Static]
+		[Export ("supportedInputContentMinScaleForDevice:")]
+		float SupportedInputContentMinScale (IMTLDevice device);
```
Comment (rolfbjarne) [inline src/metalfx.cs:297]:
```suggestion
		float GetSupportedInputContentMinScale (IMTLDevice device);
```

File: src/metalfx.cs [hunk 2]
```diff
@@ -0,0 +1,309 @@
...
+
+		[Mac (14, 0), iOS (17, 0), MacCatalyst (17, 0)]
+		[Static]
+		[Export ("supportedInputContentMaxScaleForDevice:")]
+		float SupportedInputContentMaxScale (IMTLDevice device);
```
Comment (rolfbjarne) [inline src/metalfx.cs:302]:
```suggestion
		float GetSupportedInputContentMaxScale (IMTLDevice device);
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Looks like MetalFX doesn't work on Mac Catalyst?

> ld: building for Mac Catalyst, but linking in .tbd built for macOS, file '/Applications/Xcode_15.0.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks//MetalFX.framework/MetalFX.tbd' for architecture x86_64

This may need some testing in Xcode

**Review by stephen-hawley — APPROVED** [review]

**Comment by mandel-macaque** [comment]
> Looks like MetalFX doesn't work on Mac Catalyst?
> 
> > ld: building for Mac Catalyst, but linking in .tbd built for macOS, file '/Applications/Xcode_15.0.0.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX14.0.sdk/System/Library/Frameworks//MetalFX.framework/MetalFX.tbd' for architecture x86_64
> 
> This may need some testing in Xcode

Funny enough in a m1 machine I get the following:

--- !tapi-tbd
tbd-version:     4
targets:         [ x86_64-macos, arm64-macos, arm64e-macos ]
install-name:    '/System/Library/Frameworks/MetalFX.framework/Versions/A/MetalFX'
exports:
  - targets:         [ x86_64-macos, arm64-macos, arm64e-macos ]
    objc-classes:    [ MFXSpatialScalingEffectDescriptor, MFXTemporalScalingEffectDescriptor,
                       MTLFXSpatialScalerDescriptor, MTLFXTemporalScalerDescriptor,
                       _MTLFXEffect, _MTLFXSpatialScaler, _MTLFXTemporalScaler ]
    objc-ivars:      [ _MTLFXEffect._effectID, _MTLFXEffect._encodeID, _MTLFXEffect._tracingDelegate ]
...
(END)

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**